composer/callbacks/checkpoint_saver.py

```diff
@@ -141,56 +215,63 @@ class CheckpointSaver(Callback):
 
-            See :func:`.format_name` for the available format variables.
+            The same format variables as for ``name`` are available.
 
             To disable symlinks, set this parameter to ``None``.
 
             Consider the following scenario, where:
 
-            *   The default ``save_folder='checkpoints'`` is used.
-            *   The default ``name_format='ep{epoch}-ba{batch}/rank_{rank}'`` is used.
-            *   The default ``save_latest_format='latest/rank_{rank}'`` is used.
+            *   The :attr:`~.Logger.run_name` is 'awesome-training-run'
+            *   The default ``folder='{run_name}/checkpoints'`` is used.
+            *   The default ``name='ep{epoch}-ba{batch}-rank{rank}'`` is used.
+            *   The default ``latest_filename='latest-rank{rank}'`` is used.
             *   The current epoch count is ``1``.
             *   The current batch count is ``42``.
 
-            When DeepSpeed is not being used, the rank zero process will save the checkpoint to ``'checkpoints/ep1-ba42/rank_0'``,
-            and a symlink will be created at ``'checkpoints/latest/rank_0' -> 'checkpoints/ep1-ba42/rank_0'``
+            When DeepSpeed is not being used, the rank zero process will save the checkpoint to
+            ``'awesome-training-run/checkpoints/ep1-ba42-rank0'``,
+            and a symlink will be created at
+            ``'awesome-training-run/checkpoints/latest-rank0' -> 'awesome-training-run/checkpoints/ep1-ba42-rank0'``
 
             When DeepSpeed is being used, each rank (process) will save checkpoints to::
 
-                checkpoints/ep1-ba42/rank_0.tar
-                checkpoints/ep1-ba42/rank_1.tar
-                checkpoints/ep1-ba42/rank_2.tar
+                awesome-training-run/checkpoints/ep1-ba42-rank0.tar
+                awesome-training-run/checkpoints/ep1-ba42-rank1.tar
+                awesome-training-run/checkpoints/ep1-ba42-rank2.tar
                 ...
 
             Corresponding symlinks will be created at::
 
-                checkpoints/latest/rank_0.tar -> checkpoints/ep1-ba42/rank_0.tar
-                checkpoints/latest/rank_1.tar -> checkpoints/ep1-ba42/rank_1.tar
-                checkpoints/latest/rank_2.tar -> checkpoints/ep1-ba42/rank_2.tar
+                awesome-training-run/checkpoints/latest-rank0.tar -> awesome-training-run/checkpoints/ep1-ba42-rank0.tar
+                awesome-training-run/checkpoints/latest-rank1.tar -> awesome-training-run/checkpoints/ep1-ba42-rank1.tar
+                awesome-training-run/checkpoints/latest-rank2.tar -> awesome-training-run/checkpoints/ep1-ba42-rank2.tar
                 ...
 
         overwrite (bool, optional): Whether existing checkpoints should be overridden.
-            If ``False`` (the default), then the ``checkpoint_folder`` must not exist or be empty.
+            If ``False`` (the default), then the ``folder`` must not exist or be empty.
```

Comment:
It was decided to check at training start to ensure the folder is empty, rather than check that any individual file does not exist.

The argument is named ``folder`` not ``checkpoint_folder`` for brevity (since we're already in the CheckpointSaver callback).